File: binance4j-c2c/src/main/java/com/binance4j/c2c/client/C2CMapping.java
package com.binance4j.c2c.client;

import java.util.Map;

import com.binance4j.c2c.dto.TradeHistory;
import com.binance4j.core.client.RestMapping;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.QueryMap;

/** The {@link C2CClient} mapping. */
public interface C2CMapping extends RestMapping {
	/** The base uri. */
	String BASE = "/sapi/v1/c2c/";

	/**
	 * @param map The query map.
	 * @return The generated Retrofit call.
	 */
	@GET(BASE + "orderMatch/listUserOrderHistory")
	@Headers(SIGNED_H)
	Call<TradeHistory> getTrades(@QueryMap Map<String, Object> map);
}

File: src/main/kotlin/com/binance4j/connectors/c2c/client/C2CMapping.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.connectors.c2c.client

import com.binance4j.connectors.c2c.dto.TradeHistory
import com.binance4j.connectors.core.Headers.IP_H
import com.binance4j.connectors.core.Headers.SIGNED_H
import com.binance4j.connectors.core.Headers.WEIGHT_ONE_H
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Query

/**
 * [C2CClient] mapping.
 */
interface C2CMapping {
	/**
	 * Get C2C trades.
	 *
	 * @param tradeType Trade type.
	 * @param page Result page.
	 * @param limit Result limit.
	 *
	 * @return The generated Retrofit call.
	 */
	@GET("/sapi/v1/c2c/orderMatch/listUserOrderHistory")
	@Headers(SIGNED_H, IP_H, WEIGHT_ONE_H)
	fun getTrades(@Query("tradeType") tradeType: String, @Query("page") page: Int? = null, @Query("limit") limit: Int? = null): Call<TradeHistory>
}